### PR TITLE
CB-21839 Basic secret rotation flow

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

**Comment by horadla23** [comment]
waiting for https://github.com/hortonworks/cloudbreak/pull/14870

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/ReactorFlowManager.java [hunk 1]
```diff
@@ -454,6 +461,17 @@ public void triggerDetermineDatalakeDataSizes(Long stackId, String operationId)
         reactorNotifier.notify(stackId, selector, new DetermineDatalakeDataSizesBaseEvent(selector, stackId, operationId));
     }
 
+    public FlowIdentifier triggerSecretRotation(String crn, List<SecretType> secretTypes, RotationFlowExecutionType executionType) {
+        String selector = EventSelectorUtil.selector(SecretRotationFlowChainTriggerEvent.class);
+        Optional<ResourceBasicView> resourceBasicViewByResourceCrn = stackService.getResourceBasicViewByResourceCrn(crn);
```
Comment (lturcsanyi) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/ReactorFlowManager.java:466]: I think this part belongs to StackOperationService instead and this method should get the stackId as a parameter.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java [hunk 2]
```diff
@@ -478,4 +482,15 @@ public FlowIdentifier modifyProxyConfig(NameOrCrn nameOrCrn, String accountId, S
         MDCBuilder.buildMdcContext(stack);
         return flowManager.triggerModifyProxyConfig(stack.getId(), previousProxyConfigCrn);
     }
+
+    public FlowIdentifier triggerSecretRotation(String crn, Set<String> secrets, RotationFlowExecutionType executionType) {
```
Comment (lturcsanyi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java:486]: I think this should be `List<String> secrets`, and this is also true for the request objects. (`DistroXSecretRotationRequest` and `StackV4SecretRotationRequest`)

File: flow/src/main/java/com/sequenceiq/flow/rotation/chain/SecretRotationFlowChainTriggerEvent.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public boolean equalsEvent(BaseFlowEvent other) {
+        return isClassAndEqualsEvent(SecretRotationFlowChainTriggerEvent.class, other,
+                event -> Objects.equals(secretTypes, event.secretTypes) &&
+                        Objects.equals(executionType, event.executionType));
+    }
+
+    @Override
+    public String toString() {
+        return "SecretRotationFlowChainTriggerEvent{" +
```
Comment (lturcsanyi) [inline flow/src/main/java/com/sequenceiq/flow/rotation/chain/SecretRotationFlowChainTriggerEvent.java:46]: Please add resourceCrn and resourceId

File: flow/src/main/java/com/sequenceiq/flow/rotation/event/RotationEvent.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    @Override
+    public boolean equalsEvent(BaseFlowEvent other) {
+        return isClassAndEqualsEvent(RotationEvent.class, other,
+                event -> Objects.equals(secretType, event.secretType) &&
+                        Objects.equals(executionType, event.executionType));
+    }
+
+    @Override
+    public String toString() {
+        return "RotationEvent{" +
```
Comment (lturcsanyi) [inline flow/src/main/java/com/sequenceiq/flow/rotation/event/RotationEvent.java:59]: Please add resourceCrn and resourceId

File: flow/src/main/java/com/sequenceiq/flow/rotation/service/SecretRotationService.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        return contexts;
+    }
+
+    private boolean executionNeeded(RotationFlowExecutionType inputExecutionType, RotationFlowExecutionType expectedExecutionType,
+            String resourceId, SecretType secretType) {
+        if (inputExecutionType == null || expectedExecutionType.equals(inputExecutionType)) {
+            return true;
+        }
+        LOGGER.info("Exection of {} is not needed for resource {} regarding secret {}, skipping.",
```
Comment (lturcsanyi) [inline flow/src/main/java/com/sequenceiq/flow/rotation/service/SecretRotationService.java:73]: Tipo: Exection

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java [hunk 6]
```diff
@@ -478,4 +482,18 @@ public FlowIdentifier modifyProxyConfig(NameOrCrn nameOrCrn, String accountId, S
         MDCBuilder.buildMdcContext(stack);
         return flowManager.triggerModifyProxyConfig(stack.getId(), previousProxyConfigCrn);
     }
+
+    public FlowIdentifier triggerSecretRotation(String crn, List<String> secrets, RotationFlowExecutionType executionType) {
+        StackView viewByCrn = stackDtoService.getStackViewByCrn(crn);
+        Long stackId = viewByCrn.getId();
+        List<SecretType> secretTypes = secrets.stream()
```
Comment (foldik) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java:489]: Can you please make sure that even if a secret type occurs multiple times in the list we still rotate it once or throw an error. Probably throwing an exception explicitly is better so the user can be sure about the order.

**Review by lturcsanyi — APPROVED** [review]
LGTM

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java (hunk 6 above)
Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java:489]: I think we cannot enforce any order from customer, thus I would only remove duplications, if there is any. If we need to be stricter, we can change that later.

Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java]: discussed, fixed, added UT for it

**Comment by horadla23** [comment]
rebased, added forgotten entitlement check

**Review by foldik — APPROVED** [review]
LGTM

**Pull request merged**